zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/AbstractSchema.scala

```diff
@@ -199,26 +199,26 @@ abstract class Schema(typeName: String,
 
   def getTypeName: String = typeName
   def getTimeField: TimeField
-  def copySchema: Schema
-  def asTemporal: TemporalSchema
-  def isTemporal: Boolean
+  def copySchema: AbstractSchema
+  def asSchema: Schema
```

Comment:
Yes, we can use `hasTimeField` and `asInstanceOf` to do the same thing as `asSchema`. Will remove it, but the logic is the same.